src/runners/cairoRunner.test.ts

```diff
@@ -108,6 +108,8 @@ describe('cairoRunner', () => {
         runner.run(config);
         const executionSize = runner.vm.memory.getSegmentSize(1);
         const executionEnd = runner.executionBase.add(executionSize);
+        expect(runner.vm.memory.get(executionEnd.sub(4))).toEqual(new Felt(6n));
+        expect(runner.vm.memory.get(executionEnd.sub(3))).toEqual(new Felt(14n));
         expect(runner.vm.memory.get(executionEnd.sub(2))).toEqual(new Felt(8n));
```

Comment:
The execution memory segment is ordered the same way as your `assert` statements, in your program you've first asserted `bitwise_ptr.x_and_y`, then `bitwise_ptr.x_xor_y` and finally `bitwise_ptr.x_or_y`.

Thus, between these 3 values, the one being the farthest from the `executionEnd` is `bitwise_ptr.x_and_y` (`executionEnd.sub(4)`) and the closest is `bitwise_ptr.x_or_y` (`executionEnd.sub(2)`)

Reply:
Gotcha

Reply:
Done! ✅